test/integration/puppetlabs/services/jruby/puppet_environments_int_test.clj

```diff
@@ -61,7 +61,7 @@
 ;; not ideal, but we discussed it at length and agreed that the test has
 ;; value in terms of simulating an end user's experience, which could not
 ;; be achieved w/o some such assumptions, and thus is worth keeping.
-(deftest ^:integration ^:single-threaded-only environment-flush-integration-test
```

Comment:
If this test really passes now, that's cool, I guess it sort of demonstrates that our attempts to make the environment threadsafe worked!

Reply:
Ah yeah, this still also fails in multithreaded mode, because we try to grab a catalog from specific JRubies, which isn't possible. So this tag should go back for now.

Reply:
done